blob/service.go

```diff
@@ -193,16 +202,28 @@ func (s *Service) getByCommitment(
 	height uint64,
 	namespace share.Namespace,
 	commitment Commitment,
-) (*Blob, *Proof, error) {
+) (_ *Blob, _ *Proof, err error) {
 	log.Infow("requesting blob",
 		"height", height,
 		"namespace", namespace.String())
 
+	ctx, span := tracer.Start(ctx, "get-by-commitment")
+	defer func() {
+		utils.SetStatusAndEnd(span, err)
+	}()
+	span.SetAttributes(
+		attribute.Int64("height", int64(height)),
+		attribute.String("commitment", string(commitment)),
+	)
+
 	header, err := s.headerGetter(ctx, height)
```

Comment:
We also need to measure the time it takes to get the header, so we should either wrap it with a new span within BlobService or add tracing in given header getter(preffered)

Reply:
IMO, it's better to add a trace for fetching headers inside the blob service to detect `GetByHeight` from the blob service directly. Also header.Service relies both on Syncer and Store which are in another repo. So the fastest way to add additional trace here and in case the result is confusing we can additionally cover both syncer and store with traces

Reply:
so, my suggestion to keep it :

```go
span.AddEvent("received eds", trace.WithAttributes(
		attribute.Int64("eds-size", int64(len(header.DAH.RowRoots))),
		attribute.Int64("duration", time.Since(now).Milliseconds())),
	)
```

Reply:
Agree that keeping it here would speed up things.

Reply:
Regarding keeping duration as an attribute. 
* This makes it harder to debug, and on Jaeger, you would need to look that attribute up and loose other span feature
* Attributes on traces should be treated the same way we do in metrics, where high cardinality is an issue. 

The best way here would be to wrap headerGetter with an additional span